File: dp_labs_and_solutions/original/dp_labs_and_solutions/dp_cpp_labs_and_solutions/builder/solution/journey_parser.cpp
// journey_parser.cpp

#include "journey_builder.h"
#include "journey_parser.h"

void journey_parser::parse(journey_builder & a_journey_builder)
{
	a_journey_builder.build_single_journey("Apple Computer");
	a_journey_builder.build_multiple_journey("Work", 1);
	a_journey_builder.build_single_journey("Whole Foods");
	a_journey_builder.build_single_journey("Stop & Shop");
	a_journey_builder.build_multiple_journey("Food shopping", 2);
	a_journey_builder.build_single_journey("Petco");
	a_journey_builder.build_single_journey("Stop & Shop");
	a_journey_builder.build_multiple_journey("Pet shopping", 2);
	a_journey_builder.build_multiple_journey("Today's itinerary", 3);
}

File: dp_labs_and_solutions/dp_labs_and_solutions/dp_java_labs_and_solutions/builder/lab/JourneyParser.java
// JourneyParser.java

public class JourneyParser
{
	public void parse(JourneyBuilder aJourneyBuilder)
	{
		aJourneyBuilder.buildSingleJourney("Apple Computer");
		aJourneyBuilder.buildMultipleJourney("Work", 1);
		aJourneyBuilder.buildSingleJourney("Whole Foods");
		aJourneyBuilder.buildSingleJourney("Stop & Shop");
		aJourneyBuilder.buildMultipleJourney("Food shopping", 2);
		aJourneyBuilder.buildSingleJourney("Petco");
		aJourneyBuilder.buildSingleJourney("Stop & Shop");
		aJourneyBuilder.buildMultipleJourney("Pet shopping", 2);
		aJourneyBuilder.buildMultipleJourney("Today's itinerary", 3);
	}
}
